File: src/main/java/org/meowcat/minigame/maze/Block.java
package org.meowcat.minigame.maze;

import lombok.Data;
import java.util.ArrayList;

/**
 * @author Itsusinn
 */
@Data
public class Block {

    final static int WALL = 1;
    final static int WAY = 0;
    private int type;
    private int row,column,r,c;
    private Block upWall,downWall,rightWall,leftWall;
    private boolean visited;

    public Block(int row, int column, int type){
        this.row = row;
        this.column = column;
        this.type = type;
    }

    /**
     * 获取该方块四个方向中任意一个非边界墙
     */
    public void addWall(ArrayList<Block> wallList){
        if (this.upWall != null){
            wallList.add(this.upWall);
        }
        if (this.downWall != null){
            wallList.add(this.downWall);
        }
        if (this.rightWall != null){
            wallList.add(this.rightWall);
        }
        if (this.leftWall != null){
            wallList.add(this.leftWall);
        }
    }

}

File: MazeGenerator/src/main/java/org/meowcat/minigame/maze/Block.kt
package org.meowcat.minigame.maze

import java.util.*

/**
 * @author Itsusinn
 */
data class Block(val row: Int, val column: Int, var type: Int) {
    var r = 0
    var c = 0
    var upWall: Block? = null
    var downWall: Block? = null
    var rightWall: Block? = null
    var leftWall: Block? = null
    var isVisited = false

    fun addWall(wallList: ArrayList<Block?>) {
        if (upWall != null) {
            wallList.add(upWall)
        }
        if (downWall != null) {
            wallList.add(downWall)
        }
        if (rightWall != null) {
            wallList.add(rightWall)
        }
        if (leftWall != null) {
            wallList.add(leftWall)
        }
    }

    companion object {
        const val WALL = 1
        const val WAY = 0
    }
}